tests/gnrc_sock_tcp/main.c

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (C) 2021   Simon Brummer <simon.brummer@posteo.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "shell.h"
+#include "msg.h"
+#include "net/sock/tcp.h"
+
+#define MAIN_QUEUE_SIZE (8)
+#define SOCK_TCP_QUEUE_SIZE (1)
+#define BUFFER_SIZE (1024)
+
+static msg_t main_msg_queue[MAIN_QUEUE_SIZE];
+static sock_tcp_t socks[SOCK_TCP_QUEUE_SIZE];
+static sock_tcp_t *sock = socks;
+static sock_tcp_queue_t sock_queue;
+static char buffer[BUFFER_SIZE];
+
+void dump_args(int argc, char **argv)
+{
+    printf("%s: ", argv[0]);
+    printf("argc=%d", argc);
+    for (int i = 0; i < argc; ++i) {
+        printf(", argv[%d] = %s", i, argv[i]);
+    }
+    printf("\n");
+}
+
+void print_result(const char *name, int err)
+{
+    if (err) {
+        printf("%s: returns %s\n", name, strerror(err));
+    } else {
+        printf("%s: returns %d\n", name, err);
+    }
+}
+
+int sock_tcp_connect_cmd(int argc, char **argv)
+{
+    dump_args(argc, argv);
+
+    sock_tcp_ep_t ep = SOCK_IPV6_EP_ANY;
+    gnrc_tcp_ep_from_str((gnrc_tcp_ep_t *) &ep, argv[1]);
+    uint16_t local_port = atol(argv[2]);
+    uint16_t flags = 0;
+
+    int err = sock_tcp_connect(sock, &ep, local_port, flags);
+    print_result(argv[0], err);
+    return 0;
+}
+
+int sock_tcp_disconnect_cmd(int argc, char **argv)
+{
+    dump_args(argc, argv);
+    sock_tcp_disconnect(sock);
+    printf("%s: returns\n", argv[0]);
```

Comment:
```suggestion
    sock_tcp_disconnect(sock);
```

No need to be so verbose ;)

Reply:
Then: self.child.expect_exact('sock_tcp_disconnect: returns') would never match and the test fails
I am explicitly verbose here on purpose to ensure that the tested functions actually return instead of blocking forever.